sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3281,6 +3282,22 @@ def teardown_no_lock(self,
         config = common_utils.read_yaml(handle.cluster_yaml)
         cluster_name = handle.cluster_name
         use_tpu_vm = config['provider'].get('_has_tpus', False)
+
+        # Use the new provisioner for AWS.
+        if isinstance(cloud, clouds.AWS):
```

Comment:
We also call a `ray down` in the following place, do we want to replace that as well? Or do you think that can be done in the future PR as that needs to handle the separation of `head` and `worker` nodes?

https://github.com/skypilot-org/skypilot/blob/db5d15b4269db5dbeeebdc87cca357a2fc620c00/sky/skylet/events.py#L166